opm/simulators/wells/MultisegmentWell.hpp

```diff
@@ -321,6 +321,9 @@ namespace Opm {
         // updating the inflow based on the current reservoir condition
         void updateIPR(const Simulator& ebos_simulator,
                        DeferredLogger& deferred_logger) const override;
+
+        std::tuple<typename Opm::WellInterface<TypeTag>::Scalar,typename std::decay<decltype(std::declval<decltype(std::declval<const typename Opm::WellInterface<TypeTag>::Simulator&>().model().intensiveQuantities(0, 0).fluidState())>().saltConcentration())>::type,int>
```

Comment:
this is  a really long line and hard to understand (at least for me).

Would it be possible to use the existing "using" directives to get the types and/or add more if needed?

E.g. types Scalar and FluidState are defined in this class and probably could be used.